components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2762,71 +2762,185 @@ private boolean isFilteringNotSupported(String filterOperation) {
+            String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR, 2);
+            if (attributeItems.length > 1) {
+                return createSearchValueWithDomainForCoEwOperations(attributeName, filterOperation, attributeValue,
+                        delimiter, attributeItems);
+            } else {
+                return delimiter + attributeValue + delimiter;
+            }
+        } else {
+            return delimiter + attributeValue + delimiter;
+        }
+    }
+
+    /**
+     * Create search value for EW operation.
+     *
+     * @param attributeName   Filter attribute name
+     * @param filterOperation Operator value
+     * @param attributeValue  Filter attribute value
+     * @param delimiter       Filter delimiter based on search type
+     * @return Search attribute value
+     */
+    private String createSearchValueForEwOperation(String attributeName, String filterOperation, String attributeValue,
+            String delimiter) {
+
+        // For attributes which support domain embedding, create search value by appending the delimiter after
+        // the domain separator.
+        if (isDomainSupportedAttribute(attributeName)) {
             // Extract the domain attached to the attribute value and then append the delimiter.
-            String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR);
-            if (attributeItems.length == 2) {
-                searchAttribute = attributeItems[0] + CarbonConstants.DOMAIN_SEPARATOR + delimiter + attributeItems[1];
+            String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR, 2);
+            if (attributeItems.length > 1) {
+                return createSearchValueWithDomainForCoEwOperations(attributeName, filterOperation, attributeValue,
+                        delimiter, attributeItems);
             } else {
-                searchAttribute = delimiter + attributeValue;
+                return delimiter + attributeValue;
+            }
+        } else {
+            return delimiter + attributeValue;
+        }
+    }
+
+    /**
+     * Create search value for CO and EW operations when domain is detected in the filter attribute value.
+     *
+     * @param attributeName   Filter attribute name
+     * @param filterOperation Operator value
+     * @param attributeValue  Search value
+     * @param delimiter       Filter delimiter based on search type
+     * @param attributeItems  Extracted domain and filter value
+     * @return Search attribute value
+     */
+    private String createSearchValueWithDomainForCoEwOperations(String attributeName, String filterOperation,
+            String attributeValue, String delimiter, String[] attributeItems) {
+
+        String searchAttribute;
+        if (log.isDebugEnabled()) {
+            log.debug(String.format(
+                    "Domain detected in attribute value: %s for filter attribute: %s for " + "filter operation; %s.",
```

Comment:
filter operation; -> filter operation: (change to colon)

Reply:
Fixed: https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/219